src/NuGet.Core/NuGet.Protocol/LocalRepositories/LocalPackageMetadataResource.cs

```diff
@@ -42,8 +42,7 @@ public LocalPackageMetadataResource(FindLocalPackagesResource localResource)
                 return _localResource.FindPackagesById(packageId, log, token)
                     .Where(p => includePrerelease || !p.Identity.Version.IsPrerelease)
                     .Select(GetPackageMetadata)
-                    .Select(p => metadataCache.GetObject(p))
-                    .ToList();
+                    .Select(p => metadataCache.GetObject(p));
```

Comment:
Same as the above. 

This can be super problematic in an unobvious way because it can force multiple http/file reads if the enumeration is lazified.